### ISSUE TITLE
Add the ability to set an alias to a jdk

### ISSUE THREAD

**Issue by juanmbellini** [jenv/jenv#303]
I know that jenv already creates aliases for JDKs (and that the alias option is deprecated), but I think that allowing the consumer set an alias is powerful feature. I give my own example.

I'm using graalvm and I'm testing my app using graalvm 19.3.2 (LTS) and graalvm 20.1.0.

If I want to add both JDKs, jenv reports that a given version is already present, and that the installation will be skipped.

I found a little hack which consists on renaming stuff in `$USER/.jenv/versions`, (or even adding more symlinks in order to add more aliases).

It would be nice if the consumer could configure which aliases it wants (for example, I do not want to add the alias 11, but 11.0.x), and I would like to add my own prefixes and suffixes. Maybe this could be set from a config file or something of the sort, so I could set everything there and afterwards just run something like 
```bash
$ jenv add <config-file>
```

**Comment by ctzen** [comment]
The use case I have is even simpler: `jenv add /path/to/jdk <alias-name>`

If the user add jdk with an <alias-name>, use <alias-name> instead of parsing "java -version", 
and there is no need to add the derived alias names such as openjdk64-1.8.0.262 -> 1.8.0.262 -> ... -> 1.8

Use case example. 
I have JDK 1.8.x installed and setup using jenv (for my regular java development work).
Now I would like to learn Android development, and Android Studio recommends using their JDK which is also of the 1.8 variant.
It would be nice if I can `jenv add /path/to/android-studio/jdk android-studio-jdk`
and stick a `.java-version` in the android project folder and call it a day.

**Comment by mikesaurus** [comment]
Whether it's un-deprecating the custom alias functionality or doing something new, having the ability to provide your own alias for a version is useful and I vote to keep/add this capability.

**Comment by voronaam** [comment]
The merged PR restores this functionality, albeit without undeprecating it. There are sample commands that set a custom alias in the PR's description and they should work with the latest version.

Please update and test and if it does not work for you I'll fix the problem.

**Comment by mikesaurus** [comment]
Yep, I updated earlier today and it works.  Thank you for the fix!

My reply to this post was more to support the feature request (or rather, request not to remove the feature), since deprecating the feature would imply an intention to remove it at some point and I find it to be a useful capability.

**Comment by jaimeagudo** [comment]
After the launch of Apple M1 the need for this command would increase as it would be handy to keep same versions of JDKs  on both `x86_64` and also the new `arm64e`. @voronaam , please reconsider keeping it.

**Comment by simon0191** [comment]
I installed GraalVM and the alias that was autogenerated by jenv was `64-` which is not descriptive at all.
I'd prefer to be able to set my own alias like `jenv add graalvm-java-11 graalvm-ce-java11-21.1.0 /Library/Java/JavaVirtualMachines/graalvm-ce-java11-21.1.0/Contents/Home/`

PS: Here's the output for `java -version` from GraalVM:
```
❯ java -version
openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment GraalVM CE 21.1.0 (build 11.0.11+8-jvmci-21.1-b05)
OpenJDK 64-Bit Server VM GraalVM CE 21.1.0 (build 11.0.11+8-jvmci-21.1-b05, mixed mode, sharing)
```

**Comment by simon0191** [comment]
Nevermind, I updated jenv `0.5.3 -> 0.5.4` and the alias was correctly autogenerated:
```
jenv add /Library/Java/JavaVirtualMachines/graalvm-ce-java11-21.1.0/Contents/Home/
graalvm64-11.0.11 added
11.0.11 added
11.0 added
11 added
```

**Comment by juanmbellini** [comment]
Just in case you need it, the way I'm creating aliases is by creating symlinks. Just go to `~/.jenv/versions` and execute `ls -l` and there you will know how aliases are created (they are just symlinks to each java home you have). So, if you want a new alias for a given jvm just change the name there, or create a new symlink. For instance:

``` bash
$ ln -s /Library/Java/JavaVirtualMachines/graalvm-ce-java11-20.1.0/Contents/Home ~/.jenv/versions/my-super-graal-vm-alias
```

**Comment by kid1412621** [comment]
> The merged PR restores this functionality, albeit without undeprecating it. There are sample commands that set a custom alias in the PR's description and they should work with the latest version.
> 
> Please update and test and if it does not work for you I'll fix the problem.

Any idea about the alias feature roadmap? will it be deprecated completely or came with new  implementation?

**Comment by andrewflbarnes** [comment]
I think we will undeprecate this feature - there is some (stalled) work to support multiple architectures and other aspects of versions better but the reality in this issue (and others) is that people desire or require the capability to create additional non-standard aliases.

A good example above is having some version specific to a tool ala the android studio example above or the fact that multiple graalvm distributions can have exactly the same java version
